### ISSUE TITLE
Error importing an external state object

### ISSUE THREAD

**Issue by tonisostrat** [avkonst/hookstate#233]
I seem to have run into a problem where importing external state objects fails if the initial value is a Promise. The resulting error is [HOOKSTATE-103](https://hookstate.js.org/docs/exceptions/#hookstate-103).

A reproduction is ready [here](https://codesandbox.io/s/external-state-import-zm91k). The initial view just renders a synchronous state object defined in `state.sync.js` but when you comment that import out in `fragment.js:2` and comment in `state.async.js` on the next line you get the dreaded `HOOKSTATE-103`.

Am I doing something wrong in the initialization of the asynchronous state or is this a bug?

**Comment by avkonst** [comment]
Thanks for the ticket. I can see the following stack:

```
Object.isLikelyComponentType
https://si22s.csb.app/node_modules/react-refresh/cjs/react-refresh-runtime.development.js:636:26
isReactRefreshBoundary
https://si22s.csb.app/node_modules/csbbust/refresh-helper.js:52:18
Object.postlude
https://si22s.csb.app/node_modules/csbbust/refresh-helper.js:133:13
$csb$eval
https://si22s.csb.app/src/state.async.js:13:18
J
```
It means the object is being read by the module loader... I do not know why it needs to so. Do you see the same problem with the react-scripts bundler/loader?

**Comment by tonisostrat** [comment]
The error (unfortunately) only happens in development mode. The production build works fine, it loads the data and renders the views as it's supposed to.

**Comment by avkonst** [comment]
Does it happen with react-scripts development mode?

**Comment by tonisostrat** [comment]
Sorry, yes, it's with react-scripts.

**Comment by avkonst** [comment]
Interesting... I do not have an idea how to fix it. It is the loader attempts to read the object which is not defined yet (ie. promised) and should probably handle this exception.... Do you think it can be submitted to react-scripts project?

**Comment by tonisostrat** [comment]
I will re-report it there, let's see what happens.

**Comment by avkonst** [comment]
Please let me know the ticket number. Let's see what happens, I doubt they would want to fix it so we will have to do some workaround...

**Comment by avkonst** [comment]
Document workaround for async state:
A workaround in you application would be to export a getter function instead:

```
const postsStateGlobal = createState(fetchPosts);
export function getPostsState() {
    return postsStateGlobal;

}
```

**Comment by avkonst** [comment]
Fixed in 4.0.0-rc11
